source/posix/mutex.c

```diff
@@ -0,0 +1,78 @@
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <aws/common/mutex.h>
+#include <errno.h>
+
+void aws_mutex_clean_up(struct aws_mutex *mutex) {
+    aws_mutex_release(mutex);
```

Comment:
Why are we implicitly unlocking the mutex?

Note that the OS X manpage for `pthread_mutex_unlock` documents:

> Calling pthread_mutex_unlock() with a mutex that the calling thread does not hold will result in undefined behavior.

Reply:
Will remove.